lib/matplotlib/backend_bases.py

```diff
@@ -1595,6 +1595,43 @@ def __init__(self, figure):
         self.toolbar = None  # NavigationToolbar2 will set me
         self._is_idle_drawing = False
 
+    def _repr_html_(self):
+        from base64 import b64encode
+
+        fmt = self.get_default_filetype()
+        dpi = self.figure.dpi
+        if fmt == 'retina':
+            dpi = dpi * 2
+            fmt = 'png'
+
+        kw = {
+            "format":fmt,
+            "facecolor":self.figure.get_facecolor(),
+            "edgecolor":self.figure.get_edgecolor(),
+            "dpi":dpi,
+            "bbox_inches":self.figure.bbox_inches,
+        }
+
+        bytes_io = io.BytesIO()
+        self.print_figure(bytes_io, **kw)
```

Comment:
you can just pass the kwargs here (having a separate dict doesn't seem to buy much)

Reply:
and you need to force the format here.  again, bytes_io could easily contain a postscript image here, for example.